iree/compiler/Dialect/Util/IR/UtilDialect.cpp

```diff
@@ -143,6 +143,36 @@ void UtilDialect::getCanonicalizationPatterns(
   results.insert<FoldDimOp<tensor::DimOp>>(getContext());
 }
 
+// Since all details of the interface are provided via default implementations,
+// we can just have one templated external model to apply per op, vs one
+// explicit model per op.
+template <typename OpTy>
+struct GenericNumericCastOpInterface
+    : public NumericCastOpInterface::ExternalModel<
+          GenericNumericCastOpInterface<OpTy>, OpTy> {
+  static void registerInterface(DialectRegistry &registry) {
```

Comment:
Nit: My personal preference is to not add a `static` method for the class itself. I was going to try something like 

```
template <typename T...>
void registerNumericCastOpInterface(DialectRegistry &registry);
```

then you could just do 

```
registerNumericCastOpInterface<arith::BicastOp, arith::ExtFOp, ...>(registry);
```

Advantage is if the implementation is same for all ops in dialect, then you could just get the list of all ops from tablegen generated file and use it.

Reply:
Have a look now. Never do yourself what a bit more template foo can do for you...